pkg/apis/clusteroperator/v1alpha1/types.go

```diff
@@ -416,19 +433,31 @@ type MachineSetConfig struct {
 
 	// NodeLabels specifies the labels that will be applied to nodes in this
 	// MachineSet
+	// TODO: may be obsolete as well via Cluster API selector
 	NodeLabels map[string]string `json:"nodeLabels"`
 }
 
-// MachineSetSpec is the specification for a MachineSet
+// MachineSetSpec is the Cluster Operator specification for a Cluster API machine template provider config.
+// TODO: This should be renamed, it is now used on MachineTemplate.Spec.ProviderConfig.
 type MachineSetSpec struct {
 	// MachineSetConfig is the configuration for the MachineSet
 	MachineSetConfig `json:",inline"`
 
-	// ClusterHardware specifies the hardware that the cluster will run on
+	// ClusterHardware specifies the hardware that the cluster will run on. It is typically a copy of
+	// the clusters data and set automatically by controllers.
 	ClusterHardware ClusterHardwareSpec `json:"clusterHardware"`
 
 	// ClusterVersionRef references the clusterversion the machine set is running.
 	ClusterVersionRef corev1.ObjectReference `json:"clusterVersionRef"`
+
+	// VMImage contains a single cloud provider specific image to use for this machine set.
+	VMImage VMImage `json:"vmImage"`
+}
+
+// VMImage contains a specified single image to use for a supported cloud provider.
+type VMImage struct {
+	// +optional
+	AWSImage *string `json:"awsImage"`
```

Comment:
No, just optional per API conventions, as we expand to other clouds this may or may not be set.

Reply:
Let me know if this is not correct otherwise I'll assume to leave it in.

Reply:
I suppose that `*string` is fine for now. I would expect that one of two things happen as we add more clouds. Either (1) the `*string` will be replaced by a pointer to an AWS-specific struct, that may just contain a `string`, or (2) we forego the union idea since the the VM image information is the same for every cloud and there is a one-to-one relationship between machineset and cloud.